# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/c3c/hotfix-no-macho.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- c3c-0.7.2/CMakeLists.txt.vanilla	2025-06-29 17:56:52.277084462 +0200
+++ c3c-0.7.2/CMakeLists.txt	2025-06-29 17:56:57.632170293 +0200
@@ -244,7 +244,6 @@
         find_library(LLD_COFF NAMES liblldCOFF.dylib lldCOFF.lib lldCOFF.a liblldCOFF.dll.a liblldCOFF.a PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
         find_library(LLD_COMMON NAMES liblldCommon.dylib lldCommon.lib lldCommon.a liblldCommon.dll.a liblldCommon.a PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
         find_library(LLD_ELF NAMES liblldELF.dylib lldELF.lib lldELF.a liblldELF.dll.a liblldELF.a PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
-        find_library(LLD_MACHO NAMES liblldMachO.dylib lldMachO.lib lldMachO.a liblldMachO.dll.a liblldMachO.a PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
         find_library(LLD_MINGW NAMES liblldMinGW.dylib lldMinGW.lib lldMinGW.a liblldMinGW.dll.a liblldMinGW.a PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
         find_library(LLD_WASM NAMES liblldWasm.dylib lldWasm.lib lldWasm.a liblldWasm.dll.a liblldWasm.a PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
     else()
@@ -273,7 +272,6 @@
             ${LLD_WASM}
             ${LLD_MINGW}
             ${LLD_ELF}
-            ${LLD_MACHO}
             ${LLD_COMMON}
     )
 
--- a/wrapper/src/wrapper.cpp
+++ b/wrapper/src/wrapper.cpp
@@ -49,14 +49,6 @@ namespace lld {
 		LINK_SIG
 	}
 
-	namespace mach_o {
-		LINK_SIG
-	}
-
-	namespace macho {
-		LINK_SIG
-	}
-
 	namespace wasm {
 		LINK_SIG
 	}
@@ -107,8 +99,8 @@ static bool llvm_link(ObjFormat format, const char **args, int arg_count, const
 			success = lld::elf::link(CALL_ARGS);
 			break;
 		case MACHO:
-			success = lld::macho::link(CALL_ARGS);
-			break;
+			printf("MachO not enabled");
+			exit(1);
 		case WASM:
 			success = lld::wasm::link(CALL_ARGS);
 			break;
